### PR TITLE
Add customer label parameter and a new method without prefix.

### PR DESCRIPTION
**Pull request by laikee99**
update: allow to change label every time, you can pass an empty string in the first parameter when you don't want to change label and use `client.config.Labels`.
add: add a new method `Logf` to submit a logline without prefix such as `ERROR`, `INFO` etc.

### CODE REVIEW

**Comment by afiskon** [comment]
I'm not actively maintaining this library at the moment so I would appreciate somebody else's eyes on this PR

File: promtail/protoclient.go [hunk 1]
```diff
@@ -98,33 +112,43 @@ func (c *clientProto) run() {
...
 
-func (c *clientProto) send(entries []*logproto.Entry) {
+func (c *clientProto) send(entries map[string][]*logproto.Entry) {
 	var streams []*logproto.Stream
-	streams = append(streams, &logproto.Stream{
+	/*streams = append(streams, &logproto.Stream{
 		Labels:  c.config.Labels,
 		Entries: entries,
-	})
+	})*/
```
Comment (afiskon) [inline promtail/protoclient.go]: Please don't leave any commented pieces of code.

File: promtail/jsonclient.go [hunk 2]
```diff
@@ -99,33 +112,45 @@ func (c *clientJson) run() {
...
+					level: entry.level,
+					Line:  entry.Line,
+				})
+				// batch = append(batch, entry)
 				batchSize++
 				if batchSize >= c.config.BatchEntriesNumber {
 					c.send(batch)
-					batch = []*jsonLogEntry{}
+					batch = make(map[string][]*jsonLogEntry)
+					// batch = []*jsonLogEntryLabel{}
```
Comment (afiskon) [inline promtail/jsonclient.go]: Please don't leave any commented pieces of code.

File: promtail/common.go [hunk 3]
```diff
@@ -20,6 +20,8 @@ const (
 	DISABLE LogLevel = iota
 )
 
+// 0 - 4
+
```
Comment (afiskon) [inline promtail/common.go]: Please remove this.

File: examples/client-example/main.go [hunk 4]
```diff
@@ -50,19 +50,19 @@ func main() {
...
+	labels := "{source=\"" + source_name + "\",job=\"" + job_name + "\"}"
 	conf := promtail.ClientConfig{
 		PushURL:            "http://localhost:3100/api/prom/push",
 		Labels:             labels,
 		BatchWait:          5 * time.Second,
 		BatchEntriesNumber: 10000,
-		SendLevel: 			promtail.INFO,
-		PrintLevel: 		promtail.ERROR,
+		SendLevel:          promtail.INFO,
+		PrintLevel:         promtail.ERROR,
```
Comment (afiskon) [inline examples/client-example/main.go]: I guess you executed go-fmt or something? Please submit this as a separate PR.

File: examples/client-example/main.go [hunk 5]
```diff
@@ -20,13 +20,13 @@ func displayInvalidName(arg string) {
...
-            return false
-        }
-    }
+		if !((c >= 'a' && c <= 'z') ||
+			(c >= 'A' && c <= 'Z') ||
+			(c >= '0' && c <= '9') ||
+			(c == '-') || (c == '_')) {
+			return false
+		}
+	}
```
Comment (afiskon) [inline examples/client-example/main.go]: Please remove formatting changes from the PR, or submit them as a separate PR.

**Review by afiskon — CHANGES_REQUESTED** [review]
See the comments.

**Comment by afiskon** [comment]
> you can pass an empty string in the first parameter when you don't want to change label

I don't think this is a good API, nor do I approve the break of the backward compatibility. Please provide new methods that allow to override the label if necessary and keep the existing API as is.

**Comment by laikee99** [comment]
> > you can pass an empty string in the first parameter when you don't want to change label
> 
> I don't think this is a good API, nor do I approve the break of the backward compatibility. Please provide new methods that allow to override the label if necessary and keep the existing API as is.

Agree with that, I am going to change that API and comments above. Thanks for your time.

**Comment by laikee99** [comment]
Just re-committed, please review it, thanks, I'll pull another commit with `go-formatted` later.

**Comment by laikee99** [comment]
I just tested the new method `LogfWithLabel`, it works as expected.

**Comment by olopost** [comment]
Is it possible to merge ?

**Comment by afiskon** [comment]
Unfortunately I'm not actively maintaining this library at the moment. Feel free forking it.

**Pull request closed without merging**